### PR TITLE
[APFM-89] Added comments to PR instead

### PR DESCRIPTION
**Pull request by poldz123**
### Description

It seems that the code coverage check is not properly posted to the correct workflow and in fact, a bug within the Github actions which is already posted here: [Specify check suite when creating a checkrun · Discussion #24616 · community](https://github.com/orgs/community/discussions/24616). The bug itself is 2 years old and worst case it will never be fixed by Github. 

I would recommend removing our checks and instead having it part of the job as a step and failing manually whenever the code coverage fell from 100%. This way it will always be posted to the correct workflow.

Also, whenever we do the switch to step we need to also change the check within the repo settings and remove the code-coverage check in favor of the job check.

### Solution

- Remove the create code-coverage check in the workflow
- Remove the code-coverage check within the repo settings and set a new check for the job where the new step is added.
- Create the code-coverage check as a step within the job

### Screenshot

Capture

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -51,17 +53,49 @@ async function run(): Promise<void> {
       }
     }
 
-    try {
-      const octokit = github.getOctokit(token)
-      await octokit.checks.create(createCoverageCheckRequest)
+    const octokit = new Octokit({
+        auth: token,
+    });
```
Comment (vikramatscore) [inline src/main.ts:58]: Curious about how `const octokit = github.getOctokit(token)` is different from this.

File: src/main.ts [hunk 2]
```diff
@@ -51,17 +53,49 @@ async function run(): Promise<void> {
...
+        }
+        if (!isSuccessful) {
+            const checkId = checkRequest.data.id
+            // Create comment
+            await octokit.issues.createComment({
+                ...defaultParameter,
+                issue_number: pullRequest.number,
+                body: "Uh-oh! Coverage fell: https://github.com/" + repoOwner + "/" + repoName + "/runs/" + checkId + " <!--  " + IDENTIFIER + " -->"
+            })
+        }
```
Comment (vikramatscore) [inline src/main.ts:94]: Future enhancement, we should publish an npm package with these utils.

File: src/main.ts [hunk 3]
```diff
@@ -51,17 +53,49 @@ async function run(): Promise<void> {
...
+    }
 
-      // Fail the build if there was an issue adding the check
-      core.setFailed("❌ Could not create 'Coverage check'")
+    if (!isSuccessful) {
+      core.setFailed('❌ Coverage fell')
```
Comment (vikramatscore) [inline src/main.ts:98]: Minor, how about `Coverage dropped` 😬 ?

**Review by vikramatscore — APPROVED** [review]
Looks good

File: src/main.ts (hunk 1 above)
Comment (poldz123) [reply src/main.ts]: I think using that github object to initialize octokit is a lot cleaner, shouldn't have any different I think.

Comment (poldz123) [reply src/main.ts]: Fixed

File: src/main.ts (hunk 3 above)
Comment (poldz123) [reply src/main.ts]: Fixed

**Comment by vikramatscore** [comment]
@poldz123 Do you need to recompile with `ncc` after the change in [7fec2c4](https://github.com/scoremedia/action-coverage-check/pull/260/commits/7fec2c41778846eaf5a0dffb9235ebf0b3aae4be)?

**Review by vikramatscore — APPROVED** [review]

**Pull request merged**